src/api/components/source.js

```diff
@@ -435,6 +435,8 @@ module.exports = class ComponentSource extends EntitySource {
 
             if (view) { // it is a component
                 const nameMatch = dir.name;
+                const notes = matchFile(file => anymatch(`**/${nameMatch}.${source.get('files.notes')}.md`, source._getPath(file)) );
```

Comment:
this should use the existing `isReadme` matcher - they're all abstract in the sense that they do not need to match a specific component.

Reply:
Currently, `isReadme` only match against `readme.md` and has no knowledge of the current component (cmiiw). How do we support notes as option 2 described, ie: `[component.name].[files.notes].md`?

Reply:
the matcher function is abstract, it matches files via globs.
so if currently we have `**/${this.get('files.notes')}.md` as the matcher in isReadme, we should add a second matcher `**/*.${this.get('files.notes')}.md`. Note the extra `*.`, which should now match the new file name structure.